include/gz/math/Cone.hh

```diff
@@ -0,0 +1,193 @@
+
+    // Inline bracket to help doxygen filtering.
+    inline namespace GZ_MATH_VERSION_NAMESPACE {
+    //
+    /// \class Cone Cone.hh gz/math/Cone.hh
+    /// \brief A representation of a cone.
+    ///
+    /// The cone class supports defining a cone with a radius,
+    /// length, rotational offset, and material properties. Radius and
+    /// length are in meters. See Material for more on material properties.
+    /// By default, a cone's length is aligned with the Z axis where the base
+    /// of the cone is proximal to the origin and vertex points in positive Z.
+    /// The rotational offset encodes a rotation from the z axis.
+    template<typename Precision>
+    class Cone
+    {
+      /// \brief Default constructor. The default radius and length are both
+      /// zero. The default rotational offset is
+      /// Quaternion<Precision>::Identity.
+      public: Cone() = default;
+
+      /// \brief Construct a cone with a length, radius, and optionally
+      /// a rotational offset.
+      /// \param[in] _length Length of the cone.
+      /// \param[in] _radius Radius of the cone.
+      /// \param[in] _rotOffset Rotational offset of the cone.
+      public: Cone(const Precision _length, const Precision _radius,
+                  const Quaternion<Precision> &_rotOffset =
+                  Quaternion<Precision>::Identity);
+
+      /// \brief Construct a cone with a length, radius, material and
+      /// optionally a rotational offset.
+      /// \param[in] _length Length of the cone.
+      /// \param[in] _radius Radius of the cone.
+      /// \param[in] _mat Material property for the cone.
+      /// \param[in] _rotOffset Rotational offset of the cone.
+      public: Cone(const Precision _length, const Precision _radius,
+                  const Material &_mat,
+                  const Quaternion<Precision> &_rotOffset =
+                  Quaternion<Precision>::Identity);
+
+      /// \brief Get the radius in meters.
+      /// \return The radius of the cone in meters.
+      public: Precision Radius() const;
+
+      /// \brief Set the radius in meters.
+      /// \param[in] _radius The radius of the cone in meters.
+      public: void SetRadius(const Precision _radius);
+
+      /// \brief Get the length in meters.
+      /// \return The length of the cone in meters.
+      public: Precision Length() const;
+
+      /// \brief Set the length in meters.
+      /// \param[in] _length The length of the cone in meters.
+      public: void SetLength(const Precision _length);
+
+      /// \brief Get the rotational offset. By default, a cone's length
+      /// is aligned with the Z axis. The rotational offset encodes
+      /// a rotation from the z axis.
+      /// \return The cone's rotational offset.
+      /// \sa void SetRotationalOffset(const Quaternion<Precision> &_rot)
+      public: Quaternion<Precision> RotationalOffset() const;
+
+      /// \brief Set the rotation offset.
+      /// \param[in] _rotOffset
```

Comment:
add description
```suggestion
      /// \param[in] _rotOffset
```

Reply:
This was copied directly from all the other ones, IE this doesn't even have it:
https://github.com/gazebosim/gz-math/blob/2f1e049bc6937efab3fd7a1d136978fc3339cf30/include/gz/math/Cylinder.hh#L94-L97

But I'll add  "Rotational offset quaternion"